How PWAs Drive Business Success (#2887)
GoogleChrome/web.dev: src/site/content/en/progressive-web-apps/drive-business-success/index.md

---
layout: post
title: How Progressive Web Apps can drive business success
authors:
  - sfourault
date: 2020-05-20
updated: 2020-05-20

> description: Build a solid business case for your PWA. Learn when you should invest, and how you can measure it's success.

Reviewer: ```suggestion
description: Build a solid business case for your PWA. Learn when you should invest, and how you can measure its success.
```

---

Progressive Web Apps are on a lot of companies' roadmap to modernize their
website and adapt to users' new expectations. Like all new concepts and
technical capabilities, they raise questions: is it what my customers want,
how much will it grow my business, what is technically feasible?

![Identify your stakeholders](img-1.jpg)

To shape your digital strategy, several stakeholders are often involved: the

> product manager and CMO are co-owners of the business impact of each feature,

Reviewer: ```suggestion
Product Manager and CMO are co-owners of the business impact of each feature,
```
Just keeping capitalization consistent since it's capitalized in the image and "UX Researcher" is capitalized below

the CTO assesses the feasibility and reliability of a technology, the UX
Researchers validate that a feature answers a real customer issue.

This article aims to help you answer those three questions and shape your
PWA project. You will start from your customer needs, translate this into
PWA features, and focus on measuring the business impact that each feature
brings to the table.

## PWAs solve customer needs {: #solve-customer-needs }

One rule we love to follow at Google when making products is
"[focus on the user and all else will follow][ten-things-we-know]". Think

> _user-first_, what are my customers' needs, and how does a PWA provide them?

Reviewer: ```suggestion
_user-first_: what are my customers' needs, and how does a PWA provide them?
```

![Identify customer needs](img-2.jpg)

When doing user research, we find some interesting patterns:

* Users hate delays and unreliability on mobile: the level of stress caused by
  mobile delays is [comparable to watching a horror movie][horror-movie].
* Fifty percent of smartphone users are more likely to use a company's mobile
  site when browsing or shopping because they
  [don't want to download an app][twg-shopping].
* One of the top reasons for uninstalling an app is the
  [limited storage][twg-uninstall] (whereas an installed PWA usually takes
  less than 1MB).
* Smartphone users are more likely to purchase from mobile sites that
  [offer relevant recommendations][twg-purchase] on products, and 85% of
  smartphone users say [mobile notifications are useful][twg-notifications].

According to those observations, we found out that customers prefer
experiences that are fast, installable, reliable, and engaging (F.I.R.E.)!

## PWAs leverage modern web capabilities {: #modern-capabilities }

PWAs provide a set of best practices and modern web APIs that are aimed at
meeting your customers' needs by making your site fast, installable, reliable,
and engaging.

> For example, using a Service Worker to [cache your resources][cache-resources]

Reviewer: ```suggestion
For example, using a service worker to [cache your resources][cache-resources]
```

and doing [predictive prefetching][wb-precache] makes your site faster, and
more reliable. Making your site Installable provides an easy way for your
customers to access it directly from their home screen or app launcher. And
new APIs like [Web Push Notifications][mdn-web-push] make it easier to
re-engage your users with personalized content to generate loyalty.

![Improving the user experience with new capabilities](img-3.jpg)

## Understand the business impact {: #business-impact }

The business success definition can be a lot of things depending on your
activity:

> * users spending more time on your service
> * reduced bounce rates for your leads
> * improved conversion rates
> * more returning visitors

Reviewer: ```suggestion
* Users spending more time on your service
* Reduced bounce rates for your leads
* Improved conversion rates
* More returning visitors
```

Most PWA projects result in a higher mobile conversion rate, and you can
learn more from the numerous [PWA case studies][pwa-case-studies]. Depending
on your objectives, you may want to prioritize some aspects of PWA that make
more sense for your business, and it's completely OK. PWA features can be
cherry-picked and launched separately.

Let's measure the business impact of each of these great F.I.R.E features.

### The business impact of a fast website {: #impact-fast }

A recent [study from Deloitte Digital][deloitte-study] shows that page speed
has a significant impact on business metrics.

There's a lot you can do to optimize the speed of your site to optimize the
critical user journeys for all of your users. If you don't know where to
start, take a look at our [Fast](/fast/) section, and use
[Lighthouse][lighthouse] to prioritize the most important things to fix.

> When working on your speed optimisations, start measuring your site speed
> frequently with appropriate tools and metrics to monitor your progress.
> For example, measure your Speed Index daily with [Lighthouse][lighthouse],
> fix clear targets like having a Time-to-Interactive below five seconds, and
> incorporate a [performance budget into your build process][perf-budget-ci].
> Thanks to your daily measurements and [this methodology][value-of-speed],
> you can isolate the impact of your incremental speed changes and calculate
> how much extra revenue your work has generated.

Reviewer: ```suggestion
When working on your speed optimizations, start measuring your site speed
frequently with appropriate tools and metrics to monitor your progress.
For example, measure your metrics with [Lighthouse][lighthouse],
fix clear targets like having ["Good" Core Web Vitals scores](/vitals/#core-web-vitals), and
incorporate a [performance budget into your build process][perf-budget-ci].
Thanks to your daily measurements and [the "value of speed" methodology][value-of-speed],
you can isolate the impact of your incremental speed changes and calculate
how much extra revenue your work has generated.
```

* Number of users who have successfully installed

You can start promoting your PWA install to all your users or use a more
cautious approach by only experimenting with a small group of users.

After a few days or weeks you should already have some data to measure the
impact on your business. What is the behaviour of people coming from the
installed shortcut? Do they engage more, do they convert more?

To segment users who installed your PWA, track the
[`appinstalled` event][appinstalled-event], and use JavaScript to
[check if users are in standalone mode][is-standalone] (indicating usage of
the installed PWA). Then use these as variables or dimensions for your
analytics tracking.

![Measure the value of installation](img-8.jpg)

The [case study of Weekendesk][twg-weekendest] is interesting: they propose
Installation on the second page visited to maximise the install rate, and
they observed that customers coming back via the icon on the home screen were
more than twice as likely to book a stay with them.

![Installed users had a 2.5x higher conversion rate](img-9.jpg)

Installation is a great way to make people return on your site and improve

> your customer loyalty. You can also think of personalising the experience for

Reviewer: ```suggestion
your customer loyalty. You can also think of personalizing the experience for
```

those premium users.

Even if you already have a native app, you may test to propose your app first
then push the PWA later for those who declined or did not engage with the app
install banner. Some of your users that are "semi-engaged" may not meet the
threshold for an app store-based install. This cohort can be addressed with
PWA installability that is often perceived as lighter, and with less friction.

![PWAs can reach semi-engaged users](img-10.jpg)

### The business impact of a reliable website {: #impact-reliable }

The Chrome Dino game, offered when a user is offline, is played more than
[270 million times][chrome-dino-game] a month. This impressive number shows
that network reliability is a considerable opportunity, especially in
markets with unreliable or expensive mobile data like India, Brazil, Mexico,
or Indonesia.

When a native app, installed from an app store, is launched, users expect

> it to open, regardless of whether they're connected to the Internet.

Reviewer: ```suggestion
it to open, regardless of whether they're connected to the internet.
```

Progressive Web Apps should be no different.

At a minimum, a simple offline page that tells the user the app isn't
available without a network connection should be served. Then, consider
taking the experience a step further by providing some
[functionality that makes sense while offline][pwabook-ch6]. For example,
you could provide access to tickets or boarding passes, offline wish lists,
call center contact information, articles or recipes that the user has

> recently viewed, et cetera.

Reviewer: ```suggestion
recently viewed, etc.
```

![Be helpful, even when offline](img-11.jpg)

Once you've implemented a [reliable user experience][pwabook-ch6], you may
want to measure it; how many users are going offline, in which geographies,
and do they stay on the website when the network is back?

Offline usage can be measured by recording analytics pings when the user goes
[offline or online][pwabook-ch8]. It tells you how many users continue
browsing on your website after the network comes back.

![Trivago saw a 67% of users who came back online continue browsing](img-12.jpg)

The [Trivago case study][twg-trivago] illustrates how this can impact your
business objectives: for users whose sessions were interrupted by an
offline period (around three percent of users), 67% of those who come
back online continued browsing the site.

### The business impact of an engaging website {: #impact-engaging }

Web push notifications allow users to opt-in to timely updates from sites
they love and allow you to effectively re-engage them with customized,
relevant content.

> Be careful though, asking users to sign up for web notifications when they

Reviewer: ```suggestion
Be careful, though. Asking users to sign up for web notifications when they
```

first arrive and without exposing the benefits can be perceived as spammy
and negatively affect your experience. Make sure to follow

> [best-practices][not-ux-bp] when prompting for notifications and inspire

Reviewer: ```suggestion
[best practices][not-ux-bp] when prompting for notifications and inspire
```

acceptance through relevant usages like train delays, price tracking, out

> of stock products etc.

Reviewer: ```suggestion
of stock products, etc.
```

Technically, [push notifications][dgc-web-push] on the web run in the

> background thanks to a Service Worker and are often sent by a system built for

Reviewer: ```suggestion
background thanks to a service worker and are often sent by a system built for
```

consequently sales and conversions.

To measure the effectiveness of your push campaigns, your need to measure
the whole funnel:

* Number of users eligible for push notifications
* Number of users who click a custom notification UI prompt
* Number of users who grant push notification permission
* Number of users who receive push notifications
* Number of users who engage with notifications
* Conversion and engagement of users coming from a notification

![Measure the value of Web Push Notifications](img-13.jpg)

There are a lot of great case studies on web push notifications, like
[Carrefour who multiplied their conversion rate by 4.5][carrefour-45x] by
re-engaging users with abandoned carts.

## The P in PWA: a progressive launch, feature by feature {: #feature-by-feature }

PWAs are modern websites that benefit from the massive reach of the web,
combined with all the user-friendly features that users love in native apps.
They leverage a set of best practices and modern web APIs, that can be
implemented independently depending on your business specificities and
priorities.

![Progressively launch your PWA](img-14.jpg)

> To accelerate the modernisation of your website and make it a real PWA,

Reviewer: ```suggestion
To accelerate the modernization of your website and make it a real PWA,
```

we encourage you to be agile: launch feature by feature. First, research

> with your users what features would bring them most value, then deliver

Reviewer: ```suggestion
with your users what features would bring them the most value, then deliver
```
